pkg/catalog/catalog.go

```diff
@@ -1717,50 +1718,116 @@ func (c *Catalog) PrepareExpiredCommits(ctx context.Context, repositoryID string
 	return c.Store.SaveGarbageCollectionCommits(ctx, repository, previousRunID)
 }
 
-func createParquetFile(entries []DBEntry) (string, error) {
-	// TODO: Modify using a run ID generator (https://github.com/treeverse/lakeFS/issues/4469)
-	filename := uuid.New().String()
-	fw, err := local.NewLocalFileWriter(filename)
+type GCUncommittedMark struct {
```

Comment:
We usually do `after` in the API. Here it's harder since you have 2 entities to skip. Can you document here (and later in the API) that the mark is the next key to write (and not the last key written)..?